br/pkg/lightning/backend/local/local.go

```diff
@@ -1369,207 +1126,76 @@ func (local *local) isRetryableImportTiKVError(err error) bool {
 	return common.IsRetryableError(err)
 }
 
-// writeAndIngestPairs writes the kv pairs in the range [start, end) to the peers
-// of the region, and then send the ingest command to do RocksDB ingest.
-// when return nil, it does not mean the whole task success. The success ranges is
-// recorded in the engine.finishedRanges.
-// TODO: regionSplitSize and regionSplitKeys can be a member of Engine, no need to pass it in every function.
+// writeAndIngestPairs handles a regionJob and tries to convert it to ingested stage.
+// The ingested job will record finished ranges in engine as a checkpoint.
+// If non-retryable error occurs, it will return the error.
+// If retryable error occurs, it will return nil and caller should check the stage
+// of the regionJob to determine what to do with it.
 func (local *local) writeAndIngestPairs(
 	ctx context.Context,
-	engine *Engine,
-	region *split.RegionInfo,
-	start, end []byte,
-	regionSplitSize int64,
-	regionSplitKeys int64,
+	job *regionJob,
```

Comment:
A better name is needed for this function.

Reply:
Please address this comment. Maybe we can rename it to `runJob` or `executeJob`.

Reply:
https://github.com/pingcap/tidb/pull/40692/commits/7ceb90131bef89fbb1186cc7a151290c95037cf7